Mathlib/Data/Seq/Seq.lean

```diff
@@ -510,6 +566,15 @@ def splitAt : ℕ → Seq α → List α × Seq α
       let (l, r) := splitAt n s'
       (List.cons x l, r)
 
+/-- Folds a sequence using `f`, producing sequence of intermedieate values, i.e.
+`[init, f init s.head, f (f init s.head) s.tail.head, ...]`. -/
+def fold (s : Seq α) (init : β) (f : β → α → β) : Seq β :=
```

Comment:
Would it make more sense to define this later down the file alongside all the other `fold` lemmas?

Reply:
I was trying to follow the structure of this file, where all operations on sequences are defined, and all theorems about them go below.

I think this file should be organized better, and I plan to do this in my future PRs. (I am going to add about 800 lines here in total; this is just the first part.)